examples/hosts/app-integration/external-data/tests/externalDataService.test.ts

```diff
@@ -0,0 +1,221 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { Server } from "http";
+
+import cors from "cors";
+import express from "express";
+import fetch from "node-fetch";
+import request from "supertest";
+
+import { delay } from "@fluidframework/common-utils";
+
+import { externalDataServicePort } from "../src/mock-external-data-service-interface";
+import {
+	ExternalDataSource,
+	initializeExternalDataService,
+} from "../src/mock-external-data-service";
+import { assertValidTaskData, TaskData } from "../src/model-interface";
+import { closeServer } from "./utilities";
+
+describe("mock-external-data-service", () => {
+	/**
+	 * External data source backing our service.
+	 */
+	let externalDataSource: ExternalDataSource | undefined;
+
+	/**
+	 * Express server instance backing our service.
+	 *
+	 * @remarks
+	 *
+	 * These tests spin up their own Express server instance so we can directly test against it
+	 * (using supertest), rather than leaning on network calls.
+	 */
+	let externalDataService: Server | undefined;
+
+	beforeEach(async () => {
+		externalDataSource = new ExternalDataSource();
+		externalDataService = await initializeExternalDataService({
+			port: externalDataServicePort,
+			externalDataSource,
+		});
+	});
+
+	/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+	afterEach(async () => {
+		externalDataSource = undefined;
+
+		const _externalDataService = externalDataService!;
+		externalDataService = undefined;
+
+		await closeServer(_externalDataService);
+	});
```

Comment:
We could avoid recreating the service for each test and use `/debug-reset-task-list` in `afterEach()` instead, I think? No other state that we need to clean up in the server? Could be done in a separate PR, just a potential optimization.

Reply:
Yeah, if the tests were long-running I would definitely be inclined to do that. They have seemed reasonable thus far, and doing full cleanup before each test ensures we don't get into situations where a failure in one test causes subsequent tests to fail due to bad state.